urlc98c3b97d2_Trimax_venta_tgz-pJ8-com_gesoftware_venta_structures_map_MapTestJ8/src/com/gesoftware/venta/structures/map/RemoteMap.java

```diff
@@ -0,0 +1,432 @@
+                    wait();
+            } catch (final InterruptedException e) {
+                LoggingUtility.error("Can't wait: " + e.getMessage());
+            }
+        }
+
+        if (m_Put == null)
+            return null;
+
+        final Object put = m_Put.get();
+        m_Put = null;
+
+        return (V) put;
+    } /* End of 'RemoteMap::put' method */
+
+    @Override
+    public final V remove(final Object key) {
+        if (!(key instanceof Serializable))
+            return null;
+
+        m_Puppeteer.giveCommand(new QueryRemove((Serializable) key));
+
+        synchronized (this) {
+            try {
+                while (m_Remove == null)
+                    wait();
+            } catch (final InterruptedException e) {
+                LoggingUtility.error("Can't wait: " + e.getMessage());
+            }
+        }
+
+        if (m_Remove == null)
+            return null;
+
+        final Object remove = m_Remove.get();
+        m_Remove = null;
+
+        return (V) remove;
+    } /* End of 'RemoteMap::remove' method */
+
+    @Override
+    public final void putAll(final Map<? extends K, ? extends V> m) {
+        if (!(m instanceof Serializable))
+            return;
+
+        m_Puppeteer.giveCommand(new QueryPutAll(m));
+    } /* End of 'RemoteMap::putAll' method */
+
+    @Override
+    public final void clear() {
+        m_Puppeteer.giveCommand(new QueryClear());
+    } /* End of 'RemoteMap::clear' method */
+
+    @Override
+    public final Set<K> keySet() {
+        m_Puppeteer.giveCommand(new QueryKeySet());
+
+        synchronized (this) {
+            try {
+                while (m_KeySet == null)
+                    wait();
+            } catch (final InterruptedException e) {
+                LoggingUtility.error("Can't wait: " + e.getMessage());
+            }
+        }
+
+        if (m_KeySet == null)
+            return new HashSet<K>();
+
+        final Object keySet = m_KeySet;
+        m_KeySet = null;
+
+        return (Set<K>) keySet;
+    } /* End of 'RemoteMap::keySet' method */
+
+    @Override
+    public final Collection<V> values() {
+        m_Puppeteer.giveCommand(new QueryValues());
+
+        synchronized (this) {
+            try {
+                while (m_Values == null)
+                    wait();
+            } catch (final InterruptedException e) {
+                LoggingUtility.error("Can't wait: " + e.getMessage());
+            }
+        }
+
+        if (m_Values == null)
+            return new LinkedList<V>();
+
+        final Object values = m_Values;
+        m_Values = null;
+
+        return (Collection<V>) values;
+    } /* End of 'RemoteMap::values' method */
+
+    @Override
+    public final Set<Entry<K, V>> entrySet() {
+        m_Puppeteer.giveCommand(new QueryEntrySet());
+
+        synchronized (this) {
+            try {
+                while (m_EntrySet == null)
+                    wait();
+            } catch (final InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)